internal/exp/metrics/staleness/staleness_test.go

```diff
@@ -0,0 +1,121 @@
+package staleness
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/exp/metrics/identity"
+)
+
+// RawMap
+var _ Map[time.Time] = (*RawMap[identity.Stream, time.Time])(nil)
+
+type RawMap[K comparable, V any] map[K]V
+
+func (rm *RawMap[K, V]) Load(key K) (V, bool) {
+	value, ok := (*rm)[key]
```

Comment:
given the cache-like nature, I think callers would benefit from having `Load()` initialize to zero-value when not-exist, eliminating the need for a load-check-store

Reply:
We could add `LoadOrStore(key K, value V) (V, bool)`. Which would do what you're saying.

Reply:
Added